pkg/envinfo/storage.go

```diff
@@ -166,20 +210,18 @@ func (ei *EnvInfo) GetStorageMountPath(storageName string) (string, error) {
 		return "", fmt.Errorf("invalid devfile: components.container: required value")
 	}
 
-	// since all container components have same volume mounts, we simply refer to the first container in the list
-	// refer https://github.com/openshift/odo/issues/4105 for addressing "all containers have same volume mounts"
-	paths, err := ei.devfileObj.Data.GetVolumeMountPaths(storageName, containers[0].Name)
-	if err != nil {
-		return "", err
-	}
-
-	// TODO: Below "if" condition needs to go away when https://github.com/openshift/odo/issues/4105 is addressed.
-	if len(paths) > 0 {
-		return paths[0], nil
+	// go over all containers
+	for _, c := range containers {
+		// get all volume mount paths in current container
+		pt, err := ei.devfileObj.Data.GetVolumeMountPaths(storageName, c.Name)
+		if err != nil {
+			return "", err
```

Comment:
Sorry, I missed this earlier but this will fail when the storage is mounted on the second container and not the first

```
odo storage list
The component 'nodejs-project-dmjd' has the following storage attached:
NAME      SIZE     PATH        CONTAINER     STATE
red       1Gi      /red        runtime-1     Not Pushed
red1      1Gi      /data       runtime-1     Not Pushed
red11     3Gi      /data-1     runtime-1     Not Pushed

odo storage delete red11 
 ✗  volume red11 not mounted to component runtime
```
We can continue here in case of an error.

Reply:
Fixed and also added test for this case. Pushing shortly